### PR TITLE
fix(_known_hosts_real): exclude directories from the config files

### LINKED ISSUE #817: SCP "Is a Directory"
## Describe the bug

When using completion on scp -r something/ tabtab got this error
```
++ command sed -ne 's/^[[:blank:]]*[Hh][Oo][Ss][Tt][[:blank:]]\(.*\)$/\1/p' /etc/ssh/ssh_config /home/xxxx/.ssh/config /home/xxxx/.ssh/xxxx.ssh-config/xxxx/config /home/xxxx/.ssh/xxxx.ssh-config/xxxx-xxxx/config /home/xxxx/.ssh/xxxx.ssh-config/team-xxxx/config /home/xxxx/.ssh/xxxx.ssh-config/team-xxxx/config /home/xxxx/.ssh/xxxx.ssh-config/team-xxxx/config /home/bdupuis/.ssh/xxxx.ssh-config/team-xxxx/config /home/xxxx/.ssh/local/config /home/xxxx/.ssh/local
sed: read error on /home/xxxx/.ssh/local: Is a directory
```

## To reproduce

1. Create configuration folders in .ssh
2. scp -r something_in_your_home_folder tabtab # note that scp  -r ~/something_in_your_home doesn't bug

## Expected behavior

No error message

## Versions (please complete the following information)

- [ ] Operating system name/distribution and version: 
```
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.1 LTS
Release:	22.04
Codename:	jammy
```
- [ ] bash version, `echo "$BASH_VERSION"`: 5.1.16(1)-release
- [ ] bash-completion version, `(IFS=.; echo "${BASH_COMPLETION_VERSINFO[*]}")`: 2.11

## Debug trace

```
++ command sed -ne 's/^[[:blank:]]*[Hh][Oo][Ss][Tt][[:blank:]]\(.*\)$/\1/p' /etc/ssh/ssh_config /home/xxxx/.ssh/config /home/xxxx/.ssh/xxxx.ssh-config/xxxx/config /home/xxxx/.ssh/xxxx.ssh-config/xxxx-xxxx/config /home/xxxx/.ssh/xxxx.ssh-config/team-xxxx/config /home/xxxx/.ssh/xxxx.ssh-config/team-xxxx/config /home/xxxx/.ssh/xxxx.ssh-config/team-xxxx/config /home/bdupuis/.ssh/xxxx.ssh-config/team-xxxx/config /home/xxxx/.ssh/local/config /home/xxxx/.ssh/local
sed: read error on /home/xxxx/.ssh/local: Is a directory
```

### PR DESCRIPTION
**Pull request by akinomyoga**
Fixes #817

### CODE REVIEW

File: bash_completion [hunk 1]
```diff
@@ -2023,7 +2023,7 @@ _known_hosts_real()
         for i in /etc/ssh/ssh_known_hosts /etc/ssh/ssh_known_hosts2 \
             /etc/known_hosts /etc/known_hosts2 ~/.ssh/known_hosts \
             ~/.ssh/known_hosts2; do
-            [[ -r $i ]] && kh+=("$i")
+            [[ -r $i && ! -r $i ]] && kh+=("$i")
```
Comment (scop) [inline bash_completion:2026]: I suppose

```suggestion
            [[ -r $i && ! -d $i ]] && kh+=("$i")
```

**Review by scop — CHANGES_REQUESTED** [review]
Let's add some dirs to the mix in the test suite.

File: bash_completion (hunk 1 above)
Comment (akinomyoga) [reply bash_completion]: Ah, thank you for pointing out!

**Comment by akinomyoga** [comment]
> Let's add some dirs to the mix in the test suite.

Thanks! I have added a `Include` directive in `config_include` for the test of ssh_config inclusions.

However, directories at e.g. `~/.ssh/config` and `Include <directory>` seem to be just misconfigurations. I have checked the man page `man ssh_config`, but `Include` directive does not seem to accept a directory name. I have tried it in my actual ssh configuration, but `Include <directory>` seems to be just ignored. In this sense, maybe we shouldn't fix it but leave the error messages, or we can output a more friendly message that explains the issue. But then, a question is why we have been ignoring unreadable files by testing `[[ -r $file ]]`. If we take the `Include` directive as "include-if-exists", we can consider a case an unrelated directory is created at the same path, which should be ignored.

**Comment by scop** [comment]
Parts of the discussion about error messages at https://github.com/scop/bash-completion/issues/807#issuecomment-1242687669 applies here too. Silently suppressing isn't nice, but neither is making a mess on the console, producing errors that make no sense to users, or coding and maintaining precise assumptions about undocumented details of external tool behavior into bash-completion. Whether silently suppressing is the best option or not, at least we are quite consistent with it currently. I'm open to discussing the general approach, but that's probably better done elsewhere.

**Comment by akinomyoga** [comment]
Thanks!

> Parts of the discussion about error messages at [#807 (comment)](https://github.com/scop/bash-completion/issues/807#issuecomment-1242687669) applies here too.

I also remember the discussion at https://github.com/scop/bash-completion/pull/776#issuecomment-1221233985.

> Whether silently suppressing is the best option or not, at least we are quite consistent with it currently. I'm open to discussing the general approach, but that's probably better done elsewhere.

OK! Thank you!

**Pull request merged**